### PR TITLE
core: Add a `MPKReady` so `MasterPublicKey` cannot be added afterwards

### PR DESCRIPTION
**Pull request by jimmyhu-dexon**
- `MPK` cannot be added if more than 2t+1 `MPKReady` is proposed in the same round.
- In integration_test, consensus core will be created after the network is ready.

### CODE REVIEW

File: core/configuration-chain.go [hunk 1]
```diff
@@ -126,13 +133,34 @@ func (cc *configurationChain) runDKG(round uint64) error {
...
+	for !cc.gov.IsDKGMPKReady(round) {
+		cc.logger.Info("DKG MPKs are not ready yet. Try again later...",
+			"nodeID", cc.ID)
+		cc.dkgLock.Unlock()
+		time.Sleep(500 * time.Millisecond)
+		cc.dkgLock.Lock()
+	}
 	ticker := newTicker(cc.gov, round, TickerDKG)
 	cc.dkgLock.Unlock()
 	<-ticker.Tick()
```
Comment (missionliao) [inline core/configuration-chain.go]: Does this tick necessary? I think we can call **cc.gov.DKGMasterPublicKeys** right after **cc.gov.IsDKGMPKReady() == true**

Comment (jimmyhu-dexon) [reply core/configuration-chain.go]: It's for clearing any internal tick inside ticker.

File: core/configuration-chain.go [hunk 2]
```diff
@@ -219,7 +247,7 @@ func (cc *configurationChain) runDKG(round uint64) error {
 	return nil
 }
 
-func (cc *configurationChain) isDKGReady(round uint64) bool {
+func (cc *configurationChain) isDKGFinal(round uint64) bool {
```
[the remark this answers is not in the document]
Comment (jimmyhu-dexon) [reply core/configuration-chain.go]: The loop will break out once others have proposedCRS.

File: core/test/state_test.go [hunk 3]
```diff
@@ -188,6 +202,11 @@ func (s *StateTestSuite) TestEqual() {
 	req.NoError(st.Equal(st4))
 	delete(st4.dkgMasterPublicKeys, uint64(2))
 	req.Equal(st.Equal(st4), ErrStateDKGMasterPublicKeysNotEqual)
+	// Remove dkg ready from cloned one to check if equal.
+	st4a := st.Clone()
```
[the remark this answers is not in the document]
Comment (jimmyhu-dexon) [reply core/test/state_test.go]: I just don't want to increase every `st`s below.

**Pull request merged**